### PR TITLE
assets: use MkdirAll to not error on existing dir

### LINKED ISSUE #884: bootcube doesn't allow pre-existing/mounted assets folder
**When I:**
...try to mount an existing `assets` folder into a container with `bootkube` and ask it to render assets:
```
$ docker run --rm --entrypoint=/bootkube -v $(pwd)/assets:/assets \
    quay.io/coreos/bootkube:v0.10.0 render --asset-dir=/assets
```

**Expected:**
Files created under mounted `$(pwd)/assets` folder.

**Actual:**
This output (and no files generated):
```
Error: mkdir /assets: file exists
mkdir /assets: file exists
```

...whereas if I don't mount anything into the container, it can create the folder and generate the assets successfully.

### PR DESCRIPTION
**Pull request by rphillips**
os.Mkdir will error on an existing directory. Allows for existing mount points for assets.

Fixes #884

### CODE REVIEW

**Comment by rphillips** [comment]
updated the PR and e2e failed on an AWS error

File: pkg/asset/asset.go [hunk 1]
```diff
@@ -210,9 +211,16 @@ func (as Assets) Get(name string) (Asset, error) {
...
-	if err := os.Mkdir(path, 0755); err != nil {
+	if err := os.MkdirAll(path, 0755); err != nil {
 		return err
 	}
+	files, err := ioutil.ReadDir(path)
+	if err != nil {
+		return err
+	}
+	if len(files) > 0 {
+		return errors.New("Asset directory must be empty.")
```
Comment (xiang90) [inline pkg/asset/asset.go]: Asset -> asset. Error strings should not be capitalized: https://github.com/golang/go/wiki/CodeReviewComments#error-strings

**Comment by rphillips** [comment]
coreosbot run e2e

**Comment by rphillips** [comment]
@xiang90 could you review please

**Pull request merged**